### ISSUE TITLE
iOS: dropdown does not close on blur

### ISSUE THREAD

**Issue by wujekbogdan** [shentao/vue-multiselect#709]
Normally the drop-down closes when Multiselect loses focus (in fact, not really loses focus, basically when we click outside the Multiselect element). On iOS it remains open. 

Is it a bug, or is it intentional? I would expect Multiselect to behave consistently across browsers/devices.

**Comment by TrendyTim** [comment]
Its a bug, ios doesnt seem to support div onblur

looks like it needs a variant of

```js
function hideOnClickOutside(element) {
    const outsideClickListener = event => {
        if (!element.contains(event.target)) { // or use: event.target.closest(selector) === null
            if (isVisible(element)) {
                element.style.display = 'none'
                removeClickListener()
            }
        }
    }

    const removeClickListener = () => {
        document.removeEventListener('click', outsideClickListener)
    }

    document.addEventListener('click', outsideClickListener)
}

const isVisible = elem => !!elem && !!( elem.offsetWidth || elem.offsetHeight || elem.getClientRects().length ) // source (2018-03-11): https://github.com/jquery/jquery/blob/master/src/css/hiddenVisibleSelectors.js 
```
(Taken from https://stackoverflow.com/a/3028037/4164604)

**Comment by TrendyTim** [comment]
@shentao Are there any plans to address this issue ? It really hampers the functionality on mobile.

**Comment by Michelemassari** [comment]
Any news on how to fix this? It's kind of game-breaking on iOS

**Comment by shentao** [comment]
> Although that kind of experience is a little awkward on a phone anyway. Displaying the dropdown as a full screen modal would probably work better for non mouse interaction (on smaller screens anyway).

I think this would be the best approach. Will see if I can implement this in v3. However, I don’t think this is easily done in v2.

PRs are welcome.

**Comment by Michelemassari** [comment]
I agree it's the best approach. I was wondering if there is a quick fix I can implement on currently live sites in the meantime?

Could you please help?

**Comment by spotman** [comment]
:+1: for fixing this annoying issue. Is it possible to add `close()` method to API so we can create a temporary workaround for iOS? Right now it is not possible to close vue-multiselect programmatically or to emulate `blur` event.

**Comment by KylinWu** [comment]
@spotman 

I found a method from [here](https://github.com/shentao/vue-multiselect/blob/e7133beee554f996d2aa9f791ea6e911626f8a4f/src/multiselectMixin.js#L665).

`this.$refs.multiselect.deactivate()`

**Comment by KylinWu** [comment]
Hey @spotman 

I found a method from [here](https://github.com/shentao/vue-multiselect/blob/e7133beee554f996d2aa9f791ea6e911626f8a4f/src/multiselectMixin.js#L665).

`this.$refs.reviewsFilter.deactivate()`

**Comment by KylinWu** [comment]
Hey @spotman 

I found a method from [here](https://github.com/shentao/vue-multiselect/blob/e7133beee554f996d2aa9f791ea6e911626f8a4f/src/multiselectMixin.js#L665).

`this.$refs.multiselect.deactivate()`

**Comment by spotman** [comment]
Thanks, @KylinWu ! I was really missing it.

**Comment by adunk** [comment]
@KylinWu or @spotman  Can you share a little more about how you used `this.$refs.multiselect.deactivate()` to get around this problem?
